### DISCUSSION TITLE
How to use `ixmp.utils.diff()`?

### DISCUSSION THREAD

**Discussion by cheolhung** [iiasa/message_ix#636] [MESSAGE modeling] [answered]
Dear Developer, 

I'm just wondering how to use 'ixmp utility, diff.' 
I tried 'diff' in the Westeros tutorial as attached. 
It seems that something is happening(<generator object diff at 0x000001D62A98E200>), but I have no idea how to proceed hereafter. 
As indicated in the documentation. I want to use 'diff' to identify differences in parameters between scenarios. 

![image](https://user-images.githubusercontent.com/11956390/182509822-649e0be8-0008-4a6d-acca-4e3bf2c03745.png)
 
I appreciate any help you can provide.

**Comment by khaeru** [comment, marked as answer]
Sorry for the brevity of the docs here. An example would indeed help.
- https://docs.python.org/3/tutorial/classes.html#generators is the section of the official Python tutorial on generators, with examples.
- Here is the example in the ixmp test suite of the function being used: https://github.com/iiasa/ixmp/blob/e795853ca0605f428bb423defd85f79fd8e4ab4c/ixmp/tests/test_utils.py#L38-L135

So you should do something like:
```py
# Iterate over parameter names and data
for name, data in diff(base, scen):
    print(name)
    print(data)
    # Any other operations on `name` or `data`, according to your needs
```
